src/components/settings/ARDownloadListSettings.js

```diff
@@ -0,0 +1,116 @@
+import * as FileSystem from 'expo-file-system';
+import React, { useEffect, useState } from 'react';
+import { ActivityIndicator, Alert, FlatList } from 'react-native';
+
+import { texts } from '../../config';
+import {
+  checkDownloadedData,
+  deleteAllData,
+  downloadAllData,
+  dummyData,
+  formatSize
+} from '../../helpers';
+import { ARDownloadListSettingsItem } from '../ARDownloadListSettingsItem';
+import { Button } from '../Button';
+import { SectionHeader } from '../SectionHeader';
+import { RegularText } from '../Text';
+import { Touchable } from '../Touchable';
+import { Wrapper } from '../Wrapper';
+
+const downloadAllDataAlert = (downloadAll) =>
+  Alert.alert(
+    texts.settingsTitles.arListLayouts.alertTitle,
+    texts.settingsTitles.arListLayouts.allDownloadAlertMessage,
+    [
+      {
+        text: texts.settingsTitles.arListLayouts.downloadAlertButtonText,
+        onPress: downloadAll,
+        style: 'default'
+      },
+      { text: texts.settingsTitles.arListLayouts.cancel, style: 'cancel' }
+    ]
+  );
+
+const deleteAllDataAlert = (deleteAll) =>
+  Alert.alert(
+    texts.settingsTitles.arListLayouts.alertTitle,
+    texts.settingsTitles.arListLayouts.allDeleteAlertMessage,
+    [
+      {
+        text: texts.settingsTitles.arListLayouts.deleteAlertButtonText,
+        onPress: deleteAll,
+        style: 'destructive'
+      },
+      { text: texts.settingsTitles.arListLayouts.cancel, style: 'cancel' }
+    ]
+  );
+
+const renderItem = ({ data, index, item, setData }) => (
+  <ARDownloadListSettingsItem item={item} index={index} data={data} setData={setData} />
+);
+
+export const ARDownloadListSettings = () => {
+  const [data, setData] = useState(dummyData);
+  const [loading, setLoading] = useState(false);
+  const [freeSize, setFreeSize] = useState(0);
+
+  useEffect(() => {
+    checkFreeStorage();
+  }, [data]);
+
+  useEffect(() => {
+    checkDownloadData();
+  }, []);
+
+  const checkDownloadData = async () => {
+    setLoading(false);
+    const { newDownloadedData } = await checkDownloadedData({ downloadableData: data });
+    setData(newDownloadedData);
+    setLoading(false);
```

Comment:
`setLoading(false);` twice? maybe the first one should be `true`?

Reply:
Yes, you're right!

https://github.com/ikuseiGmbH/smart-village-app-app/pull/554/commits/324ea1ff729856d5b6078dd772f5efc5b168fe53